sql/catalyst/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Relation.scala

```diff
@@ -54,15 +54,18 @@ case class DataSourceV2Relation(
 
   override lazy val metadataOutput: Seq[AttributeReference] = table match {
     case hasMeta: SupportsMetadataColumns =>
-      val resolve = conf.resolver
-      val outputNames = outputSet.map(_.name)
```

Comment:
if the goal is to let these relations report metadata cols consistently, how about we make a change here? `val outputNames = output.filterNot(_.isMetadataCol).map(_.name).toSet`.

Reply:
this is my first try, but have other issue.. see the pr description:
> And we also can not skip metadata column during filter confict name, otherwise the new generated metadata attribute will have different expr id with previous.

e.g., each call with `FileFormat.createFileMetadataCol` will generate different attribute expr id.

Reply:
how about this:
```
val metadataColFromOutput = output.filter(_.isMetadataCol)
if (metadataColFromOutput.isEmpty) {
  original code
} else {
  metadataColFromOutput
}
```

Reply:
looks good